src/deprecated.jl

```diff
@@ -12,3 +12,5 @@ function enable(; force=false)
     depwarn("`$m.enable(; force=$force)` is deprecated, use `$m.activate()` instead.", :enable)
     activate()
 end
+
+@deprecate set_active_env(f, pe) with_active_env(f, pe) false
```

Comment:
This isn't a proper deprecation as `set_active_env` would have it's changes persist beyond the scope of `f`. I think we should probably just make this a breaking change

Reply:
fair, though this is what the previous change did, modulo also renaming `set_active_env`.
That function isn't actually exported.
Since the normal way to do it is to use `apply(pe)` do.
More an internal implementation detail of `apply` ?

Reply:
I think I'm okay with stating that `set_active_env` and `get_active_env` are internal functions and don't require deprecations. Making this a breaking change would be the safest but I could be convinced to have this be non-breaking.

Reply:
If it breaks something can always revert, tag patch, and rerelease as breaking

Reply:
I will just delete them both and not deprecate